sdk/logging.go

```diff
@@ -0,0 +1,94 @@
+package sdk
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// Logger provides interface for OPA logger implementations
+type Logger interface {
+	Debug(fmt string, a ...interface{})
+	Info(fmt string, a ...interface{})
+	Error(fmt string, a ...interface{})
+	Warn(fmt string, a ...interface{})
+
+	WithFields(map[string]interface{}) Logger
+	GetFields() map[string]interface{}
+}
+
+// StandardLogger is the default OPA logger
+type StandardLogger struct {
+	fields map[string]interface{}
+}
+
+// NewStandardLogger instantiates new default OPA logger
+func NewStandardLogger() *StandardLogger {
+	return &StandardLogger{}
+}
+
+// WithFields provides additional fields to include in log output
+func (l *StandardLogger) WithFields(fields map[string]interface{}) Logger {
+	cp := l
+	cp.fields = fields
+	return cp
+}
+
+// GetFields returns additional fields of this logger
+func (l *StandardLogger) GetFields() map[string]interface{} {
+	return l.fields
+}
+
+// Debug logs at debug level
+func (l *StandardLogger) Debug(fmt string, a ...interface{}) {
+	logrus.WithFields(l.GetFields()).Debugf(fmt, a...)
```

Comment:
How would I configure the log level? E.g. if I call `l.Debug("foo %s", "bar")`, and the log level is "warning", I don't see how the level would be set on `StandardLogger`.

I think this would still be done through the logrus "global logger", a perhaps not-so-ideal situation 🤔

Reply:
Yeah that's true 🤔 Though realistically I think the logrus one will only be used when running OPA as a server, and in that case the global logger config should still apply. That notwithstanding, I agree that we should have methods for setting and reading the log level set on a logger. Will add.

Reply:
Added.